# Titanic competition with TensorFlow Decision Forests

This notebook will take you through the steps needed to train a baseline Gradient Boosted Trees Model using TensorFlow Decision Forests and creating a submission on the Titanic competition. 

This notebook shows:

1. How to do some basic pre-processing. For example, the passenger names will be tokenized, and ticket names will be splitted in parts.
1. How to train a Gradient Boosted Trees (GBT) with default parameters
1. How to train a GBT with improved default parameters
1. How to tune the parameters of a GBTs
1. How to train and ensemble many GBTs

# Imports dependencies

In [1]:
import numpy as np
import pandas as pd
import os

import tensorflow as tf
import tensorflow_decision_forests as tfdf

print(f"Found TF-DF {tfdf.__version__}")

Found TF-DF 1.2.0


# Load dataset

In [2]:
train_df = pd.read_csv("/kaggle/input/titanic/train.csv")
serving_df = pd.read_csv("/kaggle/input/titanic/test.csv")

train_df.head(10)

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S
5,6,0,3,"Moran, Mr. James",male,NaN,0,0,330877,8.4583,NaN,Q
6,7,0,1,"McCarthy, Mr. Timothy J",male,54.0,0,0,17463,51.8625,E46,S
7,8,0,3,"Palsson, Master. Gosta Leonard",male,2.0,3,1,349909,21.0750,NaN,S
8,9,1,3,"Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)",female,27.0,0,2,347742,11.1333,NaN,S
9,10,1,2,"Nasser, Mrs. Nicholas (Adele Achem)",female,14.0,1,0,237736,30.0708,NaN,C


# Prepare dataset

We will apply the following transformations on the dataset.

1. Tokenize the names. For example, "Braund, Mr. Owen Harris" will become ["Braund", "Mr.", "Owen", "Harris"].
2. Extract any prefix in the ticket. For example ticket "STON/O2. 3101282" will become "STON/O2." and 3101282.

In [3]:
def preprocess(df):
    df = df.copy()
    
    def normalize_name(x):
        return " ".join([v.strip(",()[].\"'") for v in x.split(" ")])
    
    def ticket_number(x):
        return x.split(" ")[-1]
        
    def ticket_item(x):
        items = x.split(" ")
        if len(items) == 1:
            return "NONE"
        return "_".join(items[0:-1])
    
    def get_title(name):
        title = name.split(",")[1].split(".")[0].strip()
        rare_titles = ["Lady","Countess","Capt","Col","Don","Dr",
                       "Major","Rev","Sir","Jonkheer","Dona"]
        if title in rare_titles:
            return "Rare"
        mapping = {"Mlle":"Miss","Ms":"Miss","Mme":"Mrs"}
        return mapping.get(title, title)
    
    df["Title"] = df["Name"].apply(get_title)
    df["FamilySize"] = df["SibSp"] + df["Parch"] + 1
    df["Deck"] = df["Cabin"].astype(str).str[0]
    df["FarePerPerson"] = df["Fare"] / df["FamilySize"]
    
    # ここではAge埋めをしない(後段で train の中央値をまとめて適用するため)
    
    df["Name"] = df["Name"].apply(normalize_name)
    df["Ticket_number"] = df["Ticket"].apply(ticket_number)
    df["Ticket_item"] = df["Ticket"].apply(ticket_item)
    return df


def fill_age_with_train_medians(df, age_medians_by_title, global_median):
    df = df.copy()
    df["Age"] = df.apply(
        lambda row: age_medians_by_title.get(row["Title"], global_median)
                    if pd.isna(row["Age"]) else row["Age"],
        axis=1
    )
    return df


# ① ベース前処理(Title, FamilySize, Deck, FarePerPersonまで。Ageはまだ欠損あり)
preprocessed_train_df = preprocess(train_df)
preprocessed_serving_df = preprocess(serving_df)

# ② trainだけを使って Title ごとの中央値を計算
age_medians_by_title = preprocessed_train_df.groupby("Title")["Age"].median()
global_median = preprocessed_train_df["Age"].median()  # 万一testにしかないTitleが来た場合の保険

# ③ train・test両方に、trainの中央値を適用
preprocessed_train_df = fill_age_with_train_medians(preprocessed_train_df, age_medians_by_title, global_median)
preprocessed_serving_df = fill_age_with_train_medians(preprocessed_serving_df, age_medians_by_title, global_median)

# ④ 確認
print(preprocessed_train_df["Age"].isna().sum())    # 0になっているはず
print(preprocessed_serving_df["Age"].isna().sum())  # 0になっているはず

preprocessed_train_df.head(5)

0
0


,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked,Title,FamilySize,Deck,FarePerPerson,Ticket_number,Ticket_item
0,1,0,3,Braund Mr Owen Harris,male,22.0,1,0,A/5 21171,7.2500,NaN,S,Mr,2,n,3.62500,21171,A/5
1,2,1,1,Cumings Mrs John Bradley Florence Briggs Thayer,female,38.0,1,0,PC 17599,71.2833,C85,C,Mrs,2,C,35.64165,17599,PC
2,3,1,3,Heikkinen Miss Laina,female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S,Miss,1,n,7.92500,3101282,STON/O2.
3,4,1,1,Futrelle Mrs Jacques Heath Lily May Peel,female,35.0,1,0,113803,53.1000,C123,S,Mrs,2,C,26.55000,113803,NONE
4,5,0,3,Allen Mr William Henry,male,35.0,0,0,373450,8.0500,NaN,S,Mr,1,n,8.05000,373450,NONE


Let's keep the list of the input features of the model. Notably, we don't want to train our model on the "PassengerId" and "Ticket" features.

In [4]:
input_features = list(preprocessed_train_df.columns)
input_features.remove("Ticket")
input_features.remove("PassengerId")
input_features.remove("Survived")
#input_features.remove("Ticket_number")
input_features.remove("Ticket_item")  # 負の重要度だったため除外
input_features.remove("Embarked")     # 負の重要度だったため除外
input_features.remove("Cabin")  # Deckが情報を吸収済みのため除外

print(f"Input features: {input_features}")

Input features: ['Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Fare', 'Title', 'FamilySize', 'Deck', 'FarePerPerson', 'Ticket_number']


# Convert Pandas dataset to TensorFlow Dataset

In [5]:
def tokenize_names(features, labels=None):
    """Divite the names into tokens. TF-DF can consume text tokens natively."""
    features["Name"] =  tf.strings.split(features["Name"])
    return features, labels

train_ds = tfdf.keras.pd_dataframe_to_tf_dataset(preprocessed_train_df,label="Survived").map(tokenize_names)
serving_ds = tfdf.keras.pd_dataframe_to_tf_dataset(preprocessed_serving_df).map(tokenize_names)

# Train model with default parameters

### Train model

First, we are training a GradientBoostedTreesModel model with the default parameters.

In [6]:
model = tfdf.keras.GradientBoostedTreesModel(
    verbose=0, # Very few logs
    features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
    exclude_non_specified_features=True, # Only use the features in "features"
    random_seed=1234,
)
model.fit(train_ds)

self_evaluation = model.make_inspector().evaluation()
print(f"Accuracy: {self_evaluation.accuracy} Loss:{self_evaluation.loss}")

[INFO 2026-07-05T05:08:00.346896447+00:00 kernel.cc:1214] Loading model from path /tmp/tmpt1qpgtp5/model/ with prefix 91bdfd2b565248a8
[INFO 2026-07-05T05:08:00.356465356+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T05:08:00.356542787+00:00 kernel.cc:1046] Use fast generic engine


Please report this to the TensorFlow team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output.
Cause: could not get source code
To silence this warning, decorate the function with @tf.autograph.experimental.do_not_convert
Accuracy: 0.804347813129425 Loss:0.8727316856384277


# Train model with improved default parameters

Now you'll use some specific parameters when creating the GBT model

In [7]:
model = tfdf.keras.GradientBoostedTreesModel(
    verbose=0, # Very few logs
    features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
    exclude_non_specified_features=True, # Only use the features in "features"
    
    #num_trees=2000,
    
    # Only for GBT.
    # A bit slower, but great to understand the model.
    # compute_permutation_variable_importance=True,
    
    # Change the default hyper-parameters
    # hyperparameter_template="benchmark_rank1@v1",
    
    #num_trees=1000,
    #tuner=tuner
    
    min_examples=1,
    categorical_algorithm="RANDOM",
    #max_depth=4,
    shrinkage=0.05,
    #num_candidate_attributes_ratio=0.2,
    split_axis="SPARSE_OBLIQUE",
    sparse_oblique_normalization="MIN_MAX",
    sparse_oblique_num_projections_exponent=2.0,
    num_trees=2000,
    #validation_ratio=0.0,
    random_seed=1234,
    
)
model.fit(train_ds)

self_evaluation = model.make_inspector().evaluation()
print(f"Accuracy: {self_evaluation.accuracy} Loss:{self_evaluation.loss}")

[INFO 2026-07-05T05:08:04.114954576+00:00 kernel.cc:1214] Loading model from path /tmp/tmpp8ay4_ep/model/ with prefix 71eaa96edd484b13
[INFO 2026-07-05T05:08:04.12803705+00:00 decision_forest.cc:661] Model loaded with 40 root(s), 2224 node(s), and 12 input feature(s).
[INFO 2026-07-05T05:08:04.128095547+00:00 kernel.cc:1046] Use fast generic engine


Accuracy: 0.8260869383811951 Loss:0.8949137330055237


Let's look at the model and you can also notice the information about variable importance that the model figured out

In [8]:
model.summary()

Model: "gradient_boosted_trees_model_1"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
Total params: 1
Trainable params: 0
Non-trainable params: 1
_________________________________________________________________
Type: "GRADIENT_BOOSTED_TREES"
Task: CLASSIFICATION
Label: "__LABEL"

Input Features (12):
	Age
	Deck
	FamilySize
	Fare
	FarePerPerson
	Name
	Parch
	Pclass
	Sex
	SibSp
	Ticket_number
	Title

No weights

Variable Importance: INV_MEAN_MIN_DEPTH:
    1.         "Title"  0.739661 ################
    2.           "Age"  0.271536 ##
    3. "FarePerPerson"  0.240145 #
    4.          "Fare"  0.222225 #
    5.          "Deck"  0.206450 #
    6.    "FamilySize"  0.204481 
    7.          "Name"  0.189644 
    8. "Ticket_number"  0.174218 
    9.        "Pclass"  0.171407 
   10.         "Parch"  0.170454 
   11.           "Sex"  0.170310 

Variable Importance: NUM_AS_ROOT:
    1. "Title" 37.000000 ###

In [9]:
model_with_importance = tfdf.keras.GradientBoostedTreesModel(
    verbose=0,
    features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
    exclude_non_specified_features=True,
    compute_permutation_variable_importance=True,
    random_seed=1234,
)
model_with_importance.fit(train_ds)
model_with_importance.make_inspector().variable_importances()

[INFO 2026-07-05T05:08:05.179228704+00:00 kernel.cc:1214] Loading model from path /tmp/tmpr0_ollom/model/ with prefix 2f9e5b5de281447e
[INFO 2026-07-05T05:08:05.185577769+00:00 kernel.cc:1046] Use fast generic engine


{'MEAN_DECREASE_IN_AUC_2_VS_OTHERS': [("Title" (4; #11), 0.171904761904762),
  ("Deck" (4; #1), 0.053809523809524085),
  ("Pclass" (1; #7), 0.052380952380952195),
  ("FarePerPerson" (1; #4), 0.04000000000000015),
  ("FamilySize" (1; #2), 0.015238095238095273),
  ("Age" (1; #0), 0.013571428571428346),
  ("Name" (5; #5), 0.010952380952380936),
  ("Fare" (1; #3), 0.005714285714286005),
  ("Ticket_number" (4; #10), 0.005714285714285672),
  ("Sex" (4; #8), 0.00547619047619019),
  ("Parch" (1; #6), 0.0004761904761904079),
  ("SibSp" (1; #9), 0.0)],
 'NUM_NODES': [("Name" (5; #5), 174.0),
  ("FarePerPerson" (1; #4), 118.0),
  ("Fare" (1; #3), 86.0),
  ("Age" (1; #0), 78.0),
  ("Deck" (4; #1), 75.0),
  ("Title" (4; #11), 37.0),
  ("FamilySize" (1; #2), 33.0),
  ("Pclass" (1; #7), 26.0),
  ("Ticket_number" (4; #10), 23.0),
  ("Parch" (1; #6), 9.0),
  ("Sex" (4; #8), 2.0),
  ("SibSp" (1; #9), 1.0)],
 'MEAN_DECREASE_IN_AP_2_VS_OTHERS': [("Title" (4; #11), 0.22643784482841578),
  ("Pclass" (1; #7)

# Make predictions

In [10]:
def prediction_to_kaggle_format(model, threshold=0.5):
    proba_survive = model.predict(serving_ds, verbose=0)[:,0]
    return pd.DataFrame({
        "PassengerId": serving_df["PassengerId"],
        "Survived": (proba_survive >= threshold).astype(int)
    })

def make_submission(kaggle_predictions):
    path="/kaggle/working/submission.csv"
    kaggle_predictions.to_csv(path, index=False)
    print(f"Submission exported to {path}")
    
kaggle_predictions = prediction_to_kaggle_format(model)
make_submission(kaggle_predictions)
!head /kaggle/working/submission.csv

Submission exported to /kaggle/working/submission.csv
PassengerId,Survived
892,0
893,0
894,0
895,0
896,0
897,0
898,0
899,0
900,1


# Training a model with hyperparameter tunning

Hyper-parameter tuning is enabled by specifying the tuner constructor argument of the model. The tuner object contains all the configuration of the tuner (search space, optimizer, trial and objective).


In [11]:
tuner = tfdf.tuner.RandomSearch(num_trials=1000)
tuner.choice("min_examples", [2, 5, 7, 10])
tuner.choice("categorical_algorithm", ["CART", "RANDOM"])

local_search_space = tuner.choice("growing_strategy", ["LOCAL"])
local_search_space.choice("max_depth", [3, 4, 5, 6, 8])

global_search_space = tuner.choice("growing_strategy", ["BEST_FIRST_GLOBAL"], merge=True)
global_search_space.choice("max_num_nodes", [16, 32, 64, 128, 256])

#tuner.choice("use_hessian_gain", [True, False])
tuner.choice("shrinkage", [0.02, 0.05, 0.10, 0.15])
tuner.choice("num_candidate_attributes_ratio", [0.2, 0.5, 0.9, 1.0])


tuner.choice("split_axis", ["AXIS_ALIGNED"])
oblique_space = tuner.choice("split_axis", ["SPARSE_OBLIQUE"], merge=True)
oblique_space.choice("sparse_oblique_normalization",
                     ["NONE", "STANDARD_DEVIATION", "MIN_MAX"])
oblique_space.choice("sparse_oblique_weights", ["BINARY", "CONTINUOUS"])
oblique_space.choice("sparse_oblique_num_projections_exponent", [1.0, 1.5])

# Tune the model. Notice the `tuner=tuner`.
tuned_model = tfdf.keras.GradientBoostedTreesModel(tuner=tuner)
tuned_model.fit(train_ds, verbose=0)

tuned_self_evaluation = tuned_model.make_inspector().evaluation()
print(f"Accuracy: {tuned_self_evaluation.accuracy} Loss:{tuned_self_evaluation.loss}")

Use /tmp/tmpnq4f0qjr as temporary training directory


[INFO 2026-07-05T05:10:45.86820243+00:00 kernel.cc:1214] Loading model from path /tmp/tmpnq4f0qjr/model/ with prefix 63d6da88fd9e4b96
[INFO 2026-07-05T05:10:45.885235842+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T05:10:45.885300854+00:00 kernel.cc:1046] Use fast generic engine


Accuracy: 0.8904109597206116 Loss:0.6197583675384521


In the last line in the cell above, you can see the accuracy is higher than previously with default parameters and parameters set by hand.

This is the main idea behing hyperparameter tuning.

For more information you can follow this tutorial: [Automated hyper-parameter tuning](https://www.tensorflow.org/decision_forests/tutorials/automatic_tuning_colab)

# Making an ensemble

Here you'll create 100 models with different seeds and combine their results

This approach removes a little bit the random aspects related to creating ML models

In the GBT creation is used the `honest` parameter. It will use different training examples to infer the structure and the leaf values. This regularization technique trades examples for bias estimates.

In [12]:
predictions = None
num_predictions = 0

for i in range(100):
    print(f"i:{i}")
    # Possible models: GradientBoostedTreesModel or RandomForestModel
    model = tfdf.keras.GradientBoostedTreesModel(
        verbose=0, # Very few logs
        features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
        exclude_non_specified_features=True, # Only use the features in "features"

        #min_examples=1,
        #categorical_algorithm="RANDOM",
        ##max_depth=4,
        #shrinkage=0.05,
        ##num_candidate_attributes_ratio=0.2,
        #split_axis="SPARSE_OBLIQUE",
        #sparse_oblique_normalization="MIN_MAX",
        #sparse_oblique_num_projections_exponent=2.0,
        #num_trees=2000,
        ##validation_ratio=0.0,
        random_seed=i,
        honest=True,
    )
    model.fit(train_ds)
    
    sub_predictions = model.predict(serving_ds, verbose=0)[:,0]
    if predictions is None:
        predictions = sub_predictions
    else:
        predictions += sub_predictions
    num_predictions += 1

predictions/=num_predictions

kaggle_predictions = pd.DataFrame({
        "PassengerId": serving_df["PassengerId"],
        "Survived": (predictions >= 0.5).astype(int)
    })

make_submission(kaggle_predictions)

i:0


[INFO 2026-07-05T05:10:47.138393872+00:00 kernel.cc:1214] Loading model from path /tmp/tmpblwt0x6v/model/ with prefix eccee811c0ff4b56
[INFO 2026-07-05T05:10:47.152229769+00:00 kernel.cc:1046] Use fast generic engine


i:1


[INFO 2026-07-05T05:10:48.385291941+00:00 kernel.cc:1214] Loading model from path /tmp/tmpypwfgtoz/model/ with prefix b30cbe2ebbb44f49
[INFO 2026-07-05T05:10:48.395264289+00:00 kernel.cc:1046] Use fast generic engine


i:2


[INFO 2026-07-05T05:10:49.694369651+00:00 kernel.cc:1214] Loading model from path /tmp/tmpzxtlfiwz/model/ with prefix 8cf5e241db7d43f6
[INFO 2026-07-05T05:10:49.70843124+00:00 kernel.cc:1046] Use fast generic engine


i:3


[INFO 2026-07-05T05:10:51.083531822+00:00 kernel.cc:1214] Loading model from path /tmp/tmpxrzalb9a/model/ with prefix 2c0398d8fff64da7
[INFO 2026-07-05T05:10:51.099389139+00:00 kernel.cc:1046] Use fast generic engine


i:4


[INFO 2026-07-05T05:10:52.371482977+00:00 kernel.cc:1214] Loading model from path /tmp/tmp1dcxp0l9/model/ with prefix 169481a3804245e5
[INFO 2026-07-05T05:10:52.385582318+00:00 kernel.cc:1046] Use fast generic engine


i:5


[INFO 2026-07-05T05:10:53.52442548+00:00 kernel.cc:1214] Loading model from path /tmp/tmpl527s02p/model/ with prefix 0e1adfe7a5e147f4
[INFO 2026-07-05T05:10:53.53302362+00:00 kernel.cc:1046] Use fast generic engine


i:6


[INFO 2026-07-05T05:10:54.665804537+00:00 kernel.cc:1214] Loading model from path /tmp/tmpbzjz9zjm/model/ with prefix d438b61620934f78
[INFO 2026-07-05T05:10:54.672844547+00:00 kernel.cc:1046] Use fast generic engine


i:7


[INFO 2026-07-05T05:10:55.818439972+00:00 kernel.cc:1214] Loading model from path /tmp/tmp5hpm3n2c/model/ with prefix f2434838744544ca
[INFO 2026-07-05T05:10:55.826842168+00:00 kernel.cc:1046] Use fast generic engine


i:8


[INFO 2026-07-05T05:10:57.611447289+00:00 kernel.cc:1214] Loading model from path /tmp/tmp4c67o0jo/model/ with prefix cf5e8609b9794fac
[INFO 2026-07-05T05:10:57.627401379+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T05:10:57.62746677+00:00 kernel.cc:1046] Use fast generic engine


i:9


[INFO 2026-07-05T05:10:58.777285915+00:00 kernel.cc:1214] Loading model from path /tmp/tmpc7zd0a3q/model/ with prefix a2ca10e6d0c2483b
[INFO 2026-07-05T05:10:58.784452653+00:00 kernel.cc:1046] Use fast generic engine


i:10


[INFO 2026-07-05T05:10:59.815601459+00:00 kernel.cc:1214] Loading model from path /tmp/tmpbe0x13m3/model/ with prefix 4d7260075303431b
[INFO 2026-07-05T05:10:59.821476263+00:00 kernel.cc:1046] Use fast generic engine


i:11


[INFO 2026-07-05T05:11:00.938781725+00:00 kernel.cc:1214] Loading model from path /tmp/tmphrh8u70z/model/ with prefix 081f5f8b36d848e0
[INFO 2026-07-05T05:11:00.946721493+00:00 kernel.cc:1046] Use fast generic engine


i:12


[INFO 2026-07-05T05:11:02.257927032+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgqqmsw4v/model/ with prefix bc7596583ec54b8d
[INFO 2026-07-05T05:11:02.27076583+00:00 kernel.cc:1046] Use fast generic engine


i:13


[INFO 2026-07-05T05:11:03.600681223+00:00 kernel.cc:1214] Loading model from path /tmp/tmph6n75_lj/model/ with prefix 9c7b79e5711946f9
[INFO 2026-07-05T05:11:03.615197418+00:00 kernel.cc:1046] Use fast generic engine


i:14


[INFO 2026-07-05T05:11:04.904757377+00:00 kernel.cc:1214] Loading model from path /tmp/tmp_liyw04e/model/ with prefix 4691bea1dbb649c2
[INFO 2026-07-05T05:11:04.920610443+00:00 kernel.cc:1046] Use fast generic engine


i:15


[INFO 2026-07-05T05:11:06.359773486+00:00 kernel.cc:1214] Loading model from path /tmp/tmpi_ab1qa_/model/ with prefix 270ef58b1aa74cd8
[INFO 2026-07-05T05:11:06.379209117+00:00 kernel.cc:1046] Use fast generic engine


i:16


[INFO 2026-07-05T05:11:07.49736788+00:00 kernel.cc:1214] Loading model from path /tmp/tmpuuk9uup_/model/ with prefix f345729d5ef64bb1
[INFO 2026-07-05T05:11:07.504965139+00:00 kernel.cc:1046] Use fast generic engine


i:17


[INFO 2026-07-05T05:11:08.831728784+00:00 kernel.cc:1214] Loading model from path /tmp/tmpjvs71r3d/model/ with prefix 657483a9edc24cc3
[INFO 2026-07-05T05:11:08.84444618+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T05:11:08.844493098+00:00 kernel.cc:1046] Use fast generic engine


i:18


[INFO 2026-07-05T05:11:09.961164378+00:00 kernel.cc:1214] Loading model from path /tmp/tmpofoq53rt/model/ with prefix 10eddc92333d40b6
[INFO 2026-07-05T05:11:09.968913496+00:00 kernel.cc:1046] Use fast generic engine


i:19


[INFO 2026-07-05T05:11:11.1460914+00:00 kernel.cc:1214] Loading model from path /tmp/tmp0b6l2mb0/model/ with prefix ec6b5fe797894af9
[INFO 2026-07-05T05:11:11.157358374+00:00 kernel.cc:1046] Use fast generic engine


i:20


[INFO 2026-07-05T05:11:12.342935957+00:00 kernel.cc:1214] Loading model from path /tmp/tmpe0widawu/model/ with prefix a7a05c99ff514e23
[INFO 2026-07-05T05:11:12.353349779+00:00 kernel.cc:1046] Use fast generic engine


i:21


[INFO 2026-07-05T05:11:13.552421123+00:00 kernel.cc:1214] Loading model from path /tmp/tmphgtkfgxb/model/ with prefix 6aaa7030ff804944
[INFO 2026-07-05T05:11:13.562855156+00:00 kernel.cc:1046] Use fast generic engine


i:22


[INFO 2026-07-05T05:11:14.708739872+00:00 kernel.cc:1214] Loading model from path /tmp/tmpr6pu72wy/model/ with prefix 4e64428f926b4fa0
[INFO 2026-07-05T05:11:14.717604065+00:00 kernel.cc:1046] Use fast generic engine


i:23


[INFO 2026-07-05T05:11:16.032097567+00:00 kernel.cc:1214] Loading model from path /tmp/tmphwsqf7qw/model/ with prefix 17c4bb82338748d2
[INFO 2026-07-05T05:11:16.046153854+00:00 kernel.cc:1046] Use fast generic engine


i:24


[INFO 2026-07-05T05:11:17.147845339+00:00 kernel.cc:1214] Loading model from path /tmp/tmpups8cvbc/model/ with prefix 57d8a49a275846e8
[INFO 2026-07-05T05:11:17.155431548+00:00 kernel.cc:1046] Use fast generic engine


i:25


[INFO 2026-07-05T05:11:18.617667517+00:00 kernel.cc:1214] Loading model from path /tmp/tmpuytdog2c/model/ with prefix ac82a8b954004f4f
[INFO 2026-07-05T05:11:18.635347847+00:00 kernel.cc:1046] Use fast generic engine


i:26


[INFO 2026-07-05T05:11:19.761408391+00:00 kernel.cc:1214] Loading model from path /tmp/tmp07hvjwht/model/ with prefix 11ccc31d78e144a2
[INFO 2026-07-05T05:11:19.769686474+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T05:11:19.769768367+00:00 kernel.cc:1046] Use fast generic engine


i:27


[INFO 2026-07-05T05:11:20.999587957+00:00 kernel.cc:1214] Loading model from path /tmp/tmpuzv5jkfn/model/ with prefix 42b99a91f5334646
[INFO 2026-07-05T05:11:21.009976475+00:00 kernel.cc:1046] Use fast generic engine


i:28


[INFO 2026-07-05T05:11:22.176195858+00:00 kernel.cc:1214] Loading model from path /tmp/tmpv7f72p88/model/ with prefix 5fdcedd36bc04d83
[INFO 2026-07-05T05:11:22.185097844+00:00 kernel.cc:1046] Use fast generic engine


i:29


[INFO 2026-07-05T05:11:23.820444732+00:00 kernel.cc:1214] Loading model from path /tmp/tmpi08i8sdd/model/ with prefix 4d5561f937814faf
[INFO 2026-07-05T05:11:23.826806982+00:00 kernel.cc:1046] Use fast generic engine


i:30


[INFO 2026-07-05T05:11:25.099820515+00:00 kernel.cc:1214] Loading model from path /tmp/tmpqh77y0z9/model/ with prefix b9e27259ec18426e
[INFO 2026-07-05T05:11:25.110057095+00:00 kernel.cc:1046] Use fast generic engine


i:31


[INFO 2026-07-05T05:11:26.707923772+00:00 kernel.cc:1214] Loading model from path /tmp/tmpu18tev0o/model/ with prefix 005ef83bee31469a
[INFO 2026-07-05T05:11:26.730582266+00:00 kernel.cc:1046] Use fast generic engine


i:32


[INFO 2026-07-05T05:11:27.905887357+00:00 kernel.cc:1214] Loading model from path /tmp/tmp4g1mfke5/model/ with prefix 55e77126f98f416e
[INFO 2026-07-05T05:11:27.917430093+00:00 kernel.cc:1046] Use fast generic engine


i:33


[INFO 2026-07-05T05:11:29.031788355+00:00 kernel.cc:1214] Loading model from path /tmp/tmpf5lv8562/model/ with prefix 2c5c1ebab82d4f96
[INFO 2026-07-05T05:11:29.039448379+00:00 kernel.cc:1046] Use fast generic engine


i:34


[INFO 2026-07-05T05:11:30.075413746+00:00 kernel.cc:1214] Loading model from path /tmp/tmplfpp3027/model/ with prefix 2ced72fe99b54c01
[INFO 2026-07-05T05:11:30.080527287+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T05:11:30.080579654+00:00 kernel.cc:1046] Use fast generic engine


i:35


[INFO 2026-07-05T05:11:31.146435499+00:00 kernel.cc:1214] Loading model from path /tmp/tmp_m2z9kwa/model/ with prefix ca8422ba8a534208
[INFO 2026-07-05T05:11:31.152321563+00:00 kernel.cc:1046] Use fast generic engine


i:36


[INFO 2026-07-05T05:11:32.367908504+00:00 kernel.cc:1214] Loading model from path /tmp/tmph14t2l6u/model/ with prefix 3ea426b91cc04793
[INFO 2026-07-05T05:11:32.378688765+00:00 kernel.cc:1046] Use fast generic engine


i:37


[INFO 2026-07-05T05:11:33.464038507+00:00 kernel.cc:1214] Loading model from path /tmp/tmp_phxp3qq/model/ with prefix 66353c635c8e4f26
[INFO 2026-07-05T05:11:33.470587752+00:00 kernel.cc:1046] Use fast generic engine


i:38


[INFO 2026-07-05T05:11:34.607440161+00:00 kernel.cc:1214] Loading model from path /tmp/tmpzjqmpagw/model/ with prefix b97c758af3c143a2
[INFO 2026-07-05T05:11:34.615775251+00:00 kernel.cc:1046] Use fast generic engine


i:39


[INFO 2026-07-05T05:11:35.801560616+00:00 kernel.cc:1214] Loading model from path /tmp/tmp9ogq6g8l/model/ with prefix 93c92f35c4084460
[INFO 2026-07-05T05:11:35.812680692+00:00 kernel.cc:1046] Use fast generic engine


i:40


[INFO 2026-07-05T05:11:36.94766298+00:00 kernel.cc:1214] Loading model from path /tmp/tmphoe63fc8/model/ with prefix 920d60bfbdbc452a
[INFO 2026-07-05T05:11:36.956786576+00:00 kernel.cc:1046] Use fast generic engine


i:41


[INFO 2026-07-05T05:11:38.174426825+00:00 kernel.cc:1214] Loading model from path /tmp/tmpk9qscxp5/model/ with prefix 60769f8172af41c8
[INFO 2026-07-05T05:11:38.183650106+00:00 kernel.cc:1046] Use fast generic engine


i:42


[INFO 2026-07-05T05:11:39.311933893+00:00 kernel.cc:1214] Loading model from path /tmp/tmpr9mgux9b/model/ with prefix 9ff4594cd7ea431e
[INFO 2026-07-05T05:11:39.319070562+00:00 kernel.cc:1046] Use fast generic engine


i:43


[INFO 2026-07-05T05:11:40.696508584+00:00 kernel.cc:1214] Loading model from path /tmp/tmpm649a23k/model/ with prefix a484e7316d84471d
[INFO 2026-07-05T05:11:40.713097501+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T05:11:40.713144884+00:00 kernel.cc:1046] Use fast generic engine


i:44


[INFO 2026-07-05T05:11:41.851698064+00:00 kernel.cc:1214] Loading model from path /tmp/tmpom8bk701/model/ with prefix 5271a3c0882c4d4c
[INFO 2026-07-05T05:11:41.859523234+00:00 kernel.cc:1046] Use fast generic engine


i:45


[INFO 2026-07-05T05:11:42.890329954+00:00 kernel.cc:1214] Loading model from path /tmp/tmpbxbtoutq/model/ with prefix d11f97e85c1a41bb
[INFO 2026-07-05T05:11:42.894155068+00:00 kernel.cc:1046] Use fast generic engine


i:46


[INFO 2026-07-05T05:11:44.371364291+00:00 kernel.cc:1214] Loading model from path /tmp/tmphqxrwaj_/model/ with prefix f76872bd5a164400
[INFO 2026-07-05T05:11:44.389222086+00:00 kernel.cc:1046] Use fast generic engine


i:47


[INFO 2026-07-05T05:11:45.613256903+00:00 kernel.cc:1214] Loading model from path /tmp/tmpxs_5cfta/model/ with prefix 01d07f7caa5e49a6
[INFO 2026-07-05T05:11:45.623531598+00:00 kernel.cc:1046] Use fast generic engine


i:48


[INFO 2026-07-05T05:11:46.745477225+00:00 kernel.cc:1214] Loading model from path /tmp/tmpyiuse_at/model/ with prefix 11348c68681543d3
[INFO 2026-07-05T05:11:46.753590471+00:00 kernel.cc:1046] Use fast generic engine


i:49


[INFO 2026-07-05T05:11:47.866789484+00:00 kernel.cc:1214] Loading model from path /tmp/tmp12r8vs92/model/ with prefix 1990716bee11404b
[INFO 2026-07-05T05:11:47.878316806+00:00 kernel.cc:1046] Use fast generic engine


i:50


[INFO 2026-07-05T05:11:49.07070421+00:00 kernel.cc:1214] Loading model from path /tmp/tmp2j2mr_2e/model/ with prefix d1b364712d7540be
[INFO 2026-07-05T05:11:49.079813023+00:00 kernel.cc:1046] Use fast generic engine


i:51


[INFO 2026-07-05T05:11:50.348825976+00:00 kernel.cc:1214] Loading model from path /tmp/tmp1wmrl2t2/model/ with prefix 8f2cc9ea8f09448e
[INFO 2026-07-05T05:11:50.361942673+00:00 kernel.cc:1046] Use fast generic engine


i:52


[INFO 2026-07-05T05:11:51.678611636+00:00 kernel.cc:1214] Loading model from path /tmp/tmp7nlgm2b0/model/ with prefix af1cfb2dd9134c10
[INFO 2026-07-05T05:11:51.690830674+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T05:11:51.690875984+00:00 kernel.cc:1046] Use fast generic engine


i:53


[INFO 2026-07-05T05:11:52.775526768+00:00 kernel.cc:1214] Loading model from path /tmp/tmpdl18485d/model/ with prefix 23d44cb45d3d4e17
[INFO 2026-07-05T05:11:52.78242552+00:00 kernel.cc:1046] Use fast generic engine


i:54


[INFO 2026-07-05T05:11:54.56823884+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgsotf_c6/model/ with prefix a626cf256e1b4865
[INFO 2026-07-05T05:11:54.576290472+00:00 kernel.cc:1046] Use fast generic engine


i:55


[INFO 2026-07-05T05:11:55.849597382+00:00 kernel.cc:1214] Loading model from path /tmp/tmpr0mwmg1p/model/ with prefix 738396a859b548a2
[INFO 2026-07-05T05:11:55.861076033+00:00 kernel.cc:1046] Use fast generic engine


i:56


[INFO 2026-07-05T05:11:57.184086768+00:00 kernel.cc:1214] Loading model from path /tmp/tmpchd71d7t/model/ with prefix 97de16d1786f400d
[INFO 2026-07-05T05:11:57.196889804+00:00 kernel.cc:1046] Use fast generic engine


i:57


[INFO 2026-07-05T05:11:58.425628525+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgtel9a6g/model/ with prefix 16581996d67245be
[INFO 2026-07-05T05:11:58.435522676+00:00 kernel.cc:1046] Use fast generic engine


i:58


[INFO 2026-07-05T05:11:59.615366968+00:00 kernel.cc:1214] Loading model from path /tmp/tmpzxsa7u96/model/ with prefix ddf0162515514718
[INFO 2026-07-05T05:11:59.624342276+00:00 kernel.cc:1046] Use fast generic engine


i:59


[INFO 2026-07-05T05:12:00.822394934+00:00 kernel.cc:1214] Loading model from path /tmp/tmp9giw6i3k/model/ with prefix 13f3c4397f5e470c
[INFO 2026-07-05T05:12:00.830797507+00:00 kernel.cc:1046] Use fast generic engine


i:60


[INFO 2026-07-05T05:12:02.203485615+00:00 kernel.cc:1214] Loading model from path /tmp/tmpfketp6ku/model/ with prefix a1f09d326b294194
[INFO 2026-07-05T05:12:02.216742836+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T05:12:02.216790103+00:00 kernel.cc:1046] Use fast generic engine


i:61


[INFO 2026-07-05T05:12:03.49522258+00:00 kernel.cc:1214] Loading model from path /tmp/tmpjax8vl9z/model/ with prefix 4c04d8eaa827496f
[INFO 2026-07-05T05:12:03.50867785+00:00 kernel.cc:1046] Use fast generic engine


i:62


[INFO 2026-07-05T05:12:04.70666132+00:00 kernel.cc:1214] Loading model from path /tmp/tmpokuy7obt/model/ with prefix b8338913a8ed49f0
[INFO 2026-07-05T05:12:04.717800755+00:00 kernel.cc:1046] Use fast generic engine


i:63


[INFO 2026-07-05T05:12:06.00485451+00:00 kernel.cc:1214] Loading model from path /tmp/tmp11sky1f3/model/ with prefix c82b1ae5a29e4148
[INFO 2026-07-05T05:12:06.019438764+00:00 kernel.cc:1046] Use fast generic engine


i:64


[INFO 2026-07-05T05:12:07.307510032+00:00 kernel.cc:1214] Loading model from path /tmp/tmpclp_fped/model/ with prefix eaca2a0a086f4573
[INFO 2026-07-05T05:12:07.320905911+00:00 kernel.cc:1046] Use fast generic engine


i:65


[INFO 2026-07-05T05:12:08.506793427+00:00 kernel.cc:1214] Loading model from path /tmp/tmpj1v6vb6v/model/ with prefix 9fb221302a914be9
[INFO 2026-07-05T05:12:08.514676619+00:00 kernel.cc:1046] Use fast generic engine


i:66


[INFO 2026-07-05T05:12:09.617265584+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgz_37e81/model/ with prefix ca53f8f7eb3b4491
[INFO 2026-07-05T05:12:09.623148414+00:00 kernel.cc:1046] Use fast generic engine


i:67


[INFO 2026-07-05T05:12:10.905011558+00:00 kernel.cc:1214] Loading model from path /tmp/tmp5r0rf31u/model/ with prefix 87752377cf0341b4
[INFO 2026-07-05T05:12:10.918854498+00:00 kernel.cc:1046] Use fast generic engine


i:68


[INFO 2026-07-05T05:12:12.164354737+00:00 kernel.cc:1214] Loading model from path /tmp/tmpp7htrmth/model/ with prefix e5a81e05bd344e00
[INFO 2026-07-05T05:12:12.176641741+00:00 kernel.cc:1046] Use fast generic engine


i:69


[INFO 2026-07-05T05:12:13.748979283+00:00 kernel.cc:1214] Loading model from path /tmp/tmpe461tqyw/model/ with prefix d3f8abd061ca4f43
[INFO 2026-07-05T05:12:13.769684237+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T05:12:13.769755544+00:00 kernel.cc:1046] Use fast generic engine


i:70


[INFO 2026-07-05T05:12:15.272405918+00:00 kernel.cc:1214] Loading model from path /tmp/tmppwuilg6s/model/ with prefix 8ed7810273d44ffb
[INFO 2026-07-05T05:12:15.29062501+00:00 kernel.cc:1046] Use fast generic engine


i:71


[INFO 2026-07-05T05:12:16.493992943+00:00 kernel.cc:1214] Loading model from path /tmp/tmpn9df_bva/model/ with prefix 657772b6a5a2410f
[INFO 2026-07-05T05:12:16.504105645+00:00 kernel.cc:1046] Use fast generic engine


i:72


[INFO 2026-07-05T05:12:17.789337452+00:00 kernel.cc:1214] Loading model from path /tmp/tmp89pi4ivj/model/ with prefix 76637d8fd5544cc2
[INFO 2026-07-05T05:12:17.802319729+00:00 kernel.cc:1046] Use fast generic engine


i:73


[INFO 2026-07-05T05:12:19.065041718+00:00 kernel.cc:1214] Loading model from path /tmp/tmpeu0zk2cv/model/ with prefix ffe1461a0a964e5b
[INFO 2026-07-05T05:12:19.076358427+00:00 kernel.cc:1046] Use fast generic engine


i:74


[INFO 2026-07-05T05:12:20.349993341+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgghmkf5p/model/ with prefix 599c07d76cfa4a75
[INFO 2026-07-05T05:12:20.360079048+00:00 kernel.cc:1046] Use fast generic engine


i:75


[INFO 2026-07-05T05:12:21.565253016+00:00 kernel.cc:1214] Loading model from path /tmp/tmp40kasies/model/ with prefix ad3872903b2b4c8f
[INFO 2026-07-05T05:12:21.573634402+00:00 kernel.cc:1046] Use fast generic engine


i:76


[INFO 2026-07-05T05:12:22.61093895+00:00 kernel.cc:1214] Loading model from path /tmp/tmpuw86k8z9/model/ with prefix 83b183ed606b4ef6
[INFO 2026-07-05T05:12:22.614858827+00:00 kernel.cc:1046] Use fast generic engine


i:77


[INFO 2026-07-05T05:12:23.681823027+00:00 kernel.cc:1214] Loading model from path /tmp/tmpdyh4ktdo/model/ with prefix 3706448467a146de
[INFO 2026-07-05T05:12:23.687470584+00:00 kernel.cc:1046] Use fast generic engine


i:78


[INFO 2026-07-05T05:12:24.77946742+00:00 kernel.cc:1214] Loading model from path /tmp/tmpsbk7ua9b/model/ with prefix 4cf6b9917bc04a16
[INFO 2026-07-05T05:12:24.784921308+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T05:12:24.784981168+00:00 kernel.cc:1046] Use fast generic engine


i:79


[INFO 2026-07-05T05:12:25.999712706+00:00 kernel.cc:1214] Loading model from path /tmp/tmpk31ws57d/model/ with prefix 3e57b9ec00494c04
[INFO 2026-07-05T05:12:26.008886526+00:00 kernel.cc:1046] Use fast generic engine


i:80


[INFO 2026-07-05T05:12:27.938178651+00:00 kernel.cc:1214] Loading model from path /tmp/tmpqn3k_y3k/model/ with prefix 154ea459b1854902
[INFO 2026-07-05T05:12:27.947905347+00:00 kernel.cc:1046] Use fast generic engine


i:81


[INFO 2026-07-05T05:12:29.231521959+00:00 kernel.cc:1214] Loading model from path /tmp/tmpf_t6986z/model/ with prefix 7d716a8845eb4e42
[INFO 2026-07-05T05:12:29.240764045+00:00 kernel.cc:1046] Use fast generic engine


i:82


[INFO 2026-07-05T05:12:30.827116692+00:00 kernel.cc:1214] Loading model from path /tmp/tmp1blml88m/model/ with prefix 2f4672d78312483a
[INFO 2026-07-05T05:12:30.848198516+00:00 kernel.cc:1046] Use fast generic engine


i:83


[INFO 2026-07-05T05:12:32.213407311+00:00 kernel.cc:1214] Loading model from path /tmp/tmpmm47wuh9/model/ with prefix d2a436c3af2f487a
[INFO 2026-07-05T05:12:32.225003803+00:00 kernel.cc:1046] Use fast generic engine


i:84


[INFO 2026-07-05T05:12:33.378460903+00:00 kernel.cc:1214] Loading model from path /tmp/tmphqi7jtt6/model/ with prefix 57c610bbff9543e3
[INFO 2026-07-05T05:12:33.385695562+00:00 kernel.cc:1046] Use fast generic engine


i:85


[INFO 2026-07-05T05:12:34.44415923+00:00 kernel.cc:1214] Loading model from path /tmp/tmpes09w4fm/model/ with prefix e5e9910061f845c9
[INFO 2026-07-05T05:12:34.449180848+00:00 kernel.cc:1046] Use fast generic engine


i:86


[INFO 2026-07-05T05:12:35.599493466+00:00 kernel.cc:1214] Loading model from path /tmp/tmppfarn6_j/model/ with prefix 140c155ec93b402c
[INFO 2026-07-05T05:12:35.608620229+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T05:12:35.608677402+00:00 kernel.cc:1046] Use fast generic engine


i:87


[INFO 2026-07-05T05:12:36.935495002+00:00 kernel.cc:1214] Loading model from path /tmp/tmpv9d_2wvk/model/ with prefix 50be5c2a0bed4f4a
[INFO 2026-07-05T05:12:36.950776198+00:00 kernel.cc:1046] Use fast generic engine


i:88


[INFO 2026-07-05T05:12:38.511185744+00:00 kernel.cc:1214] Loading model from path /tmp/tmpm94j4mjy/model/ with prefix e1c52f0aac2c4ad7
[INFO 2026-07-05T05:12:38.533759102+00:00 kernel.cc:1046] Use fast generic engine


i:89


[INFO 2026-07-05T05:12:39.724421838+00:00 kernel.cc:1214] Loading model from path /tmp/tmpriowvazp/model/ with prefix c0d2d5c14e5a4a14
[INFO 2026-07-05T05:12:39.733980187+00:00 kernel.cc:1046] Use fast generic engine


i:90


[INFO 2026-07-05T05:12:40.821722662+00:00 kernel.cc:1214] Loading model from path /tmp/tmpk3dztcty/model/ with prefix ed1b006685714263
[INFO 2026-07-05T05:12:40.829812062+00:00 kernel.cc:1046] Use fast generic engine


i:91


[INFO 2026-07-05T05:12:41.943690778+00:00 kernel.cc:1214] Loading model from path /tmp/tmpbmd05k52/model/ with prefix 7d200525eb76443f
[INFO 2026-07-05T05:12:41.95286343+00:00 kernel.cc:1046] Use fast generic engine


i:92


[INFO 2026-07-05T05:12:43.807952196+00:00 kernel.cc:1214] Loading model from path /tmp/tmpstshu1du/model/ with prefix 7fac4dc0f3a24cdf
[INFO 2026-07-05T05:12:43.840521692+00:00 kernel.cc:1046] Use fast generic engine


i:93


[INFO 2026-07-05T05:12:45.090367671+00:00 kernel.cc:1214] Loading model from path /tmp/tmp84i1jcqk/model/ with prefix bafaca815f364cd9
[INFO 2026-07-05T05:12:45.102229389+00:00 kernel.cc:1046] Use fast generic engine


i:94


[INFO 2026-07-05T05:12:46.180563822+00:00 kernel.cc:1214] Loading model from path /tmp/tmpg3xszbwy/model/ with prefix 8026e387b6ca4b3e
[INFO 2026-07-05T05:12:46.18754512+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T05:12:46.187585801+00:00 kernel.cc:1046] Use fast generic engine


i:95


[INFO 2026-07-05T05:12:47.301704303+00:00 kernel.cc:1214] Loading model from path /tmp/tmpfr4z49ji/model/ with prefix 24df565ae4874d3e
[INFO 2026-07-05T05:12:47.310979797+00:00 kernel.cc:1046] Use fast generic engine


i:96


[INFO 2026-07-05T05:12:48.472023754+00:00 kernel.cc:1214] Loading model from path /tmp/tmpxitdh2yw/model/ with prefix 3d6ff06454ea4602
[INFO 2026-07-05T05:12:48.479617204+00:00 kernel.cc:1046] Use fast generic engine


i:97


[INFO 2026-07-05T05:12:49.555226802+00:00 kernel.cc:1214] Loading model from path /tmp/tmpiqw8sude/model/ with prefix 231f3b787f1f41e4
[INFO 2026-07-05T05:12:49.565682268+00:00 kernel.cc:1046] Use fast generic engine


i:98


[INFO 2026-07-05T05:12:50.767490695+00:00 kernel.cc:1214] Loading model from path /tmp/tmp5z5mhp5q/model/ with prefix 57afecbf7cdf4949
[INFO 2026-07-05T05:12:50.776973918+00:00 kernel.cc:1046] Use fast generic engine


i:99


[INFO 2026-07-05T05:12:51.881452938+00:00 kernel.cc:1214] Loading model from path /tmp/tmph4x_nr6x/model/ with prefix cd521453dfd64ef5
[INFO 2026-07-05T05:12:51.889386988+00:00 kernel.cc:1046] Use fast generic engine


Submission exported to /kaggle/working/submission.csv
